### DISCUSSION TITLE
In Select, how to filter options based on category?

### DISCUSSION THREAD

**Discussion by DevMushref** [filamentphp/filament#6125] [Help] [answered]
I have a Model "Observation Types":

![image](https://user-images.githubusercontent.com/25293489/230036469-7f63dc4e-6de1-42e1-82c9-3d543fd9c94b.png)

Each type is set in a category "Safety" or "Security". In another Resource form, there is a `Select` field that fetches these types through relationship.

```
Forms\Components\Select::make('observation_type_relationship_id')
                        ->searchable()
                        ->label('Observation type')
                        ->preload()
                        ->relationship('observationTypeRelationship', 'name'),
```

And a `Radio` field that has 2 options if this is a security or a safety observation:

```
Forms\Components\Radio::make('observation')
                                ->label('what is the observation?')
                                ->reactive()
                                ->options([
                                    1 => 'safety',
                                    0 => 'security',
                                ]),
```

How do I filter the Select field based on the radio selection? If "Security" is chosen, then show only observation types labeled "security" (As shown in the image above).

**Comment by valpuia** [comment]
https://filamentphp.com/docs/2.x/tables/filters#getting-started

https://filamentphp.com/docs/2.x/tables/filters#relationship-select-filters

**Reply by DevMushref** [reply]
Thanks, but I want that in the form not the table.

**Comment by danharrin** [comment]
Use the third parameter of `relationship()` to scope the `$query` based on `$get('observation')`

**Reply by DevMushref** [reply]
Thank you for your response, could you please give an example?

**Reply by danharrin** [reply, marked as answer]
` ->relationship('observationTypeRelationship', 'name', fn ($query, $get) => $query->where('observation', $get('observation')))`

**Reply by aleirsyad** [reply]
Thanks!
